File: backend/src/main/java/com/viniciusfinger/rsflood/model/HomeStructure.java
package com.viniciusfinger.rsflood.model;

import com.viniciusfinger.rsflood.enums.FloorMaterial;
import com.viniciusfinger.rsflood.enums.HomeMaterial;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class HomeStructure {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private HomeMaterial material;

    @Enumerated(EnumType.STRING)
    private FloorMaterial floorMaterial;

    //Caixa d'água
    private Boolean hasWaterTank;

    //Água encanada
    private Boolean hasWaterPipe;

    //Fossa séptica
    private Boolean hasSepticTank;

    private Integer roomsNumber;

    private Integer bathroomsNumber;

}

File: frontend/src/app/model/home-structure.ts
import { FloorMaterial } from "./enum/floor-material";
import { HomeMaterial } from "./enum/home-material";

export class HomeStructure {
  material: HomeMaterial;
  floorMaterial: FloorMaterial;
  hasWaterTank: boolean;
  hasWaterPipe: boolean;
  hasSepticTank: boolean;
  roomsNumber: number;
  bathroomsNumber: number;

  constructor(
    material: HomeMaterial,
    floorMaterial: FloorMaterial,
    hasWaterTank: boolean,
    hasWaterPipe: boolean,
    hasSepticTank: boolean,
    roomsNumber: number,
    bathroomsNumber: number
  ) {
    this.material = material;
    this.floorMaterial = floorMaterial;
    this.hasWaterTank = hasWaterTank;
    this.hasWaterPipe = hasWaterPipe;
    this.hasSepticTank = hasSepticTank;
    this.roomsNumber = roomsNumber;
    this.bathroomsNumber = bathroomsNumber;
  }
}
